runtime/runtime/src/prefetch.rs

```diff
@@ -142,10 +142,17 @@ impl TriePrefetcher {
                 }
             }
 
+            let mut code_prefetch_requested = false;
```

Comment:
My quick reading of the prefetcher has led me to understanding that it will not make duplicated prefetches if requested to prefetch the same key multiple times. Is that not the case?

Reply:
From what I see, we will read the in-memory data prefetched before. Still, I am not comfortable with sending many requests for the same code, even if processing them is fast.